pkg/kv/kvpb/api.proto

```diff
@@ -2200,6 +2165,47 @@ message AddSSTableResponse {
   bytes following_likely_non_empty_span_start = 4 [(gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/roachpb.Key"];
 }
 
+message LinkExternalSSTableRequest {
+    RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+
+  // SSTTimestampToRequestTimestamp gives the timestamp used for all MVCC keys
+  // in the provided SST. If this timestamp differs from the request timestamp
+  // (e.g. if the request gets pushed) then all MVCC keys in the SST will be
+  // rewritten to the request timestamp during request evaluation. This ensures
+  // the writes comply with the timestamp cache and closed timestamp. It also
+  // causes the AddSSTable to be emitted via the range feed, since it respects
+  // the closed timestamp.
+  //
+  // Callers should always set this, except in very special circumstances when
+  // the timestamp cache and closed timestamp can safely be ignored (e.g.
+  // streaming replication into an offline tenant).
+  //
+  // Note that this alone is not sufficient to guarantee MVCC correctness, since
+  // it can write below or replace existing committed versions (the tscache is
+  // only bumped when the values are subsequently read).
+  util.hlc.Timestamp sst_timestamp_to_request_timestamp = 2 [
+    (gogoproto.customname) = "SSTTimestampToRequestTimestamp", (gogoproto.nullable) = false];
+
+  // MVCCStats, if set, is the MVCCStats for the contents of this SSTable and is
+  // used as-is during evaluation of the command to update the range
+  // MVCCStats, instead of computing the stats for the SSTable by iterating it.
+  // Including these stats can make the evaluation of LinkExternalSSTable much cheaper.
+  storage.enginepb.MVCCStats mvcc_stats = 3 [(gogoproto.customname) = "MVCCStats"];
+
+  message  ExternalFile {
```

Comment:
Thoughts on  either:
- flattening this proto (i.e. get rid of the `ExternalFile` message)? 
- or, we could use the `ExternalFile` message to only describe the backing file (i.e. we'd remove the `SyntheticPrefix` field) and use this message in the `proposer_kv.proto`.

Reply:
yeah, I'd flatten it.

Reply:
well, wait, no, we likely want to be able to repeat external files when we restore inc backups. I'd keep it as is.